jscatter/_cli.py

```diff
@@ -0,0 +1,62 @@
+import argparse
+import os
+import shutil
+import sys
+from pathlib import Path
+
+from jscatter import __version__
+
+_DEV = False
+
+
+def check_uv_available():
+    if shutil.which("uv") is None:
+        print("Error: 'uv' command not found.", file=sys.stderr)
+        print("Please install 'uv' to run `jscatter demo` entrypoint.", file=sys.stderr)
+        print(
+            "For more information, visit: https://github.com/astral-sh/uv",
+            file=sys.stderr,
+        )
+        sys.exit(1)
+
+
+def run_notebook(notebook_path: Path):
+    check_uv_available()
+
+    command = [
+        "uvx",
+        "--python",
+        "3.12",
+        "--from",
+        "jupyter-core",
+        "--with",
+        "jupyterlab",
+        "--with",
+        "." if _DEV else f"jupyter-scatter=={__version__}",
+        "jupyter",
+        "lab",
+        str(notebook_path),
+    ]
+
+    try:
+        os.execvp(command[0], command)
```

Comment:
maybe for a follow-up PR, but process substitution only works on unix systems. This will fail on windows.

Reply:
Does anybody still use windows? 🤔🤭🤫

Thanks for bringing this up! Is this the same issue that came up in `juv`?

Reply:
> Does anybody still use windows? 🤔🤭🤫

Lots!

> Thanks for bringing this up! Is this the same issue that came up in juv?

Yeah, it also came up in juv. It's a non-trivial problem, especially with Jupyter. The solutions I've found require a `subprocess`, but the issue is that you also need to intercept keyboard interrupts to gracefully exit. Also I'm not sure the executable is called `uv` on windows (e.g., `uv.exe`).  

Here's an example from `uv`:

- https://github.com/astral-sh/uv/blob/main/python/uv/__main__.py

They delegate whether to use `subprocess` or `execvpe` based on `sys.platform`. In `juv` we ended up just using `subprocess` in either case:

- https://github.com/manzt/juv/blob/main/src/juv/_run_replace.py

Reply:
Thanks for sharing! I ended up implementing a strategy similar to `uv`'s. Hopefully it's working for windows folks.